packages/strapi-design-system/src/NumberInput/NumberInput.js

```diff
@@ -149,13 +160,13 @@ export const NumberInput = React.forwardRef(
     };
 
     const handleFocus = () => {
-      if (value !== undefined) {
+      if (value !== undefined && value !== null) {
```

Comment:
Just checked this case. Its not possible to achieve null in strapi due to [this code](https://github.com/strapi/strapi/blob/b0d73be3e776b4dfde0251b8befe5a4a63a960f8/packages/core/helper-plugin/lib/src/components/GenericInput/index.js#L52). In strapi component will always have `undefined` instead of `null`.

Reply:
After spending some time on research i got following.

`NumberInput` is used 3 times in strapi/strapi repo:

1. [`GenericInput`](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/GenericInput/index.js) with [`const value = defaultValue ?? undefined;`](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/GenericInput/index.js#L52) so the value always be **undefined**.
2. [`Inputs`](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js) which is used only in [`FilterPopoverURLQuery`](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/index.js). `Inputs` component is not handling nullish values itself but `FilterPopoverURLQuery` default value is empty string ([default state](**url**), [value passed directly to `Inputs`](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/index.js#L167)). Also you should notice that you got two components named `Inputs` but this one is used only in `FilterPopoverURLQuery`. You can check it by typing path like this in global project search `FilterPopoverURLQuery/Inputs` because there is no any other way to import `Inputs` component from _FilterPopoverURLQuery folder_.
3. [`CheckboxWithNumberField`](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js) with [0 value by default](https://github.com/strapi/strapi/blob/master/packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js#L83).


**Conclusion**.
Seems like there is no way that `NumberInput` will get `null` value passed into props. I`m not part of core team and can make a mistake and I provided full information so you can check this out by yourself fast enough.

Reply:
Thanks for checking. I agree with your analysis and also think it will never receive `null`. If so, it is a bug.